src/Models/ShallowWaterModels/shallow_water_advection_operators.jl

```diff
@@ -9,13 +10,28 @@ function div_hUv(i, j, k, grid, advection, solution)
 end
 
 @inline momentum_flux_huu(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uu(i, j, k, grid, solution.uh, solution.uh) / h[i, j, k]
+    @inbounds momentum_flux_uu(i, j, k, grid, advection, solution.uh, solution.uh) / solution.h[i, j, k]
 
 @inline momentum_flux_huv(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_uv(i, j, k, grid, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
+    @inbounds momentum_flux_uv(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
 @inline momentum_flux_hvu(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_vu(i, j, k, grid, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, h)
+    @inbounds momentum_flux_vu(i, j, k, grid, advection, solution.uh, solution.vh) / ℑxyᶠᶠᵃ(i, j, k, grid, solution.h)
 
 @inline momentum_flux_hvv(i, j, k, grid, advection, solution) =
-    @inbounds momentum_flux_vv(i, j, k, grid, solution.vh, solution.vh) / h[i, j, k]
+    @inbounds momentum_flux_vv(i, j, k, grid, advection, solution.vh, solution.vh) / solution.h[i, j, k]
+
+
+
+function div_UV(i, j, k, grid, solution)
+
+    return 1 / Vᵃᵃᶜ(i, j, k, grid) * (δxᶜᵃᵃ(i, j, k, grid, mass_flux_x, solution.uh) +
+                                      δyᵃᶜᵃ(i, j, k, grid, mass_flux_y, solution.vh))
+end
+
+@inline mass_flux_x(i, j, k, grid, uh) = @inbounds Ax_ψᵃᵃᶠ(i, j, k, grid, uh)
+
+@inline mass_flux_y(i, j, k, grid, vh) = @inbounds Ay_ψᵃᵃᶠ(i, j, k, grid, vh) 
+
+
+
```

Comment:
Bit of extra whitespace down here.

Reply:
Echoing an above comment: could make sense to save fixing `h` interpolation for a future PR.

Reply:
Removed